### PR TITLE
MULE-15434: Add metadata category and resolver name to components

### PR DESCRIPTION
**Pull request by alegmarra**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java [hunk 1]
```diff
@@ -152,9 +175,10 @@ private void enrichWithDsql(OperationDeclaration declaration, MethodElement meth
                                                                                                                        .nativeOutputResolver(),
                                                                                                                    query
                                                                                                                        .entityResolver())));
+
       addQueryModelProperties(declaration, query);
       declareDynamicType(declaration.getOutput());
-      declareMetadataKeyId(declaration, new NullMetadataScopeAdapter());
+      declareMetadataKeyId(declaration, null);
```
Comment (juandesi) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java]: is this ok?

Comment (juandesi) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java]: I mean the “null”

Comment (alegmarra) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java]: sadly, it is, we use null everywhere to mark a non-existing categoryName

Comment (juandesi) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/DynamicMetadataDeclarationEnricher.java]: is this allowed in the model? when is the category name absent?

**Pull request merged**